mobile_verifier/src/radio_threshold.rs

```diff
@@ -255,16 +255,8 @@ where
         &self,
         report: &RadioThresholdReportReq,
     ) -> anyhow::Result<RadioThresholdReportVerificationStatus> {
-        let is_legacy = self.verify_legacy(&report.hotspot_pubkey, &None).await?;
         let report_validity = self.do_report_verifications(report).await;
-        let final_validity = if is_legacy
-            && report_validity == RadioThresholdReportVerificationStatus::ThresholdReportStatusValid
-        {
-            RadioThresholdReportVerificationStatus::ThresholdReportStatusLegacyValid
-        } else {
-            report_validity
-        };
-        Ok(final_validity)
+        Ok(report_validity)
```

Comment:
This function can no longer fail.

Reply:
As I see it can fail in case the pubkey is unknown (RadioThresholdReportVerificationStatus::ThresholdReportStatusInvalidCarrierKey) 
![image](https://github.com/user-attachments/assets/42963959-f7f5-437a-aa01-b43fbc62e9a6)

Reply:
Good catch, fixed